x-pack/platform/plugins/shared/task_manager/server/task_store.ts

```diff
@@ -440,6 +578,15 @@ export class TaskStore {
    * @returns {Promise<void>}
    */
   public async remove(id: string): Promise<void> {
+    const taskInstance = await this.get(id);
+
+    if (taskInstance.userScope?.apiKey && !taskInstance.userScope?.apiKeyCreatedByUser) {
+      const apiKeyId = Buffer.from(taskInstance.userScope.apiKey, 'base64')
+        .toString()
+        .split(':')[0];
+      this.security.authc.apiKeys.invalidateAsInternalUser({ ids: [apiKeyId] });
```

Comment:
looks like this is missing an await? should we catch any errors that might be. thrown from this?

Reply:
Feels like this should have set off a lint error ...